src/ORM/Query.php

```diff
@@ -717,6 +717,10 @@ public function count()
             return (int)$counter($query);
         }
 
+        if (!empty($this->_cache)) {
```

Comment:
This isn't right, but no. The call to [QueryTrait::all](https://github.com/cakephp/cakephp/blob/ad-cache-count-too/src/Datasource/QueryTrait.php#L213-L215) would populate and return the cached value.